lms/services/application_instance.py

```diff
@@ -186,6 +188,19 @@ def update_application_instance(  # pylint:disable=too-many-arguments
             application_instance.aes_cipher_iv = aes_iv
             application_instance.developer_secret = encrypted_secret
 
+        if organization_public_id:
+            org = self._organization_service.get_by_public_id(organization_public_id)
+            if org:
+                application_instance.organization = org
+            else:
+                raise ValidationError(
+                    messages={
+                        "organization_public_id": [
+                            f"Organization {organization_public_id} not found"
+                        ]
+                    }
+                )
```

Comment:
I'm not sure if it's weird to raise a `ValidationError` here? How do we know this came from a field called `organization_public_id` in this context?